src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -198,6 +208,54 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
             executionContext.Debug("Diagnostic file upload complete.");
         }
 
+        // Dumping Agent Azure VM extension logs to the support files folder.
+        private void DumpAgentExtensionLogs(IExecutionContext executionContext, string supportFilesFolder, DateTime jobStartTimeUtc)
+        {
+            executionContext.Debug("Starting dumping agent extension logs.");
+
+            string pathToLogs = null;
+            string archiveName = null;
+            string timestamp = jobStartTimeUtc.ToString("yyyyMMdd-HHmmss");
+
+            if (PlatformUtil.RunningOnWindows)
+            {
+                string pathToVersions = "C:\\WindowsAzure\\Logs\\Plugins\\Microsoft.VisualStudio.Services.TeamServicesAgent";
+                if (Directory.Exists(pathToVersions))
+                {
+                    string[] subDirs = Directory.GetDirectories(pathToVersions).Select(dir => dir.Split("\\").Last()).ToArray();
+                    Version[] versions = subDirs.Select(dir => new Version(dir)).ToArray();
+                    Version maxVersion = versions.Max();
+                    pathToLogs = Path.Combine(pathToVersions, maxVersion.ToString());
```

Comment:
Just to clarify: I originally thought that we will need to apply this filtering for both Windows and Linux systems. Does this mean that on Linux the extension doesn't create a subfolder with a version name? If this is a case, could you please add code comments about this difference?